.circleci/config.yml

```
@@ -112,11 +112,52 @@ jobs:
             include_visit_job_action: false
             failure_message: '${SLACK_FAILURE_MESSAGE}'
             success_message: '${SLACK_SUCCESS_MESSAGE}'
+  UI Tests:
+    # Props to work done by others in this file and in
+    # https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/.circleci/config.yml
+    # this is a combined rip-off from both
+    parameters:     
+      device:
+        description: The device (e.g. "iPhone 12") to use when running tests.
+        type: string
+        default: "iPhone 12"
+    executor:
+      name: ios/default
+      xcode-version: "12.4.0"
+    environment:
+      FL_OUTPUT_DIR: output
+      FASTLANE_LANE: run_UI_tests
+    steps:
+      - git/shallow-checkout
+      - ios/install-dependencies:
+            bundle-install: true
+            pod-install: true
+      - run:
+          name: Copy Secrets
+          command: bundle exec fastlane run configure_apply
+      # Start: Swift Package Manager Workaround
+      # Ref. https://support.circleci.com/hc/en-us/articles/360044709573-Swift-Package-Manager-fails-to-clone-from-private-Git-repositories
+      - run: rm ~/.ssh/id_rsa
+      - run: for ip in $(dig @8.8.8.8 github.com +short); do ssh-keyscan github.com,$ip; ssh-keyscan $ip; done 2>/dev/null >> ~/.ssh/known_hosts || true
+      # END: Swift Package Manager Workaround
+      - run:
+          name: Fastlane
+          command: bundle exec fastlane $FASTLANE_LANE device:'<< parameters.device >>'
```

Putting the name of the lane in an environment variable adds a layer of indirection here: to know the exact command we have to find the value of the env var.

Is there a particular reason for it? If not, I'd suggest using the value directly:

```suggestion
          command: bundle exec fastlane run_UI_tests device:'<< parameters.device >>'
```

---

Yes, there was no real reason to do this. I think it sneaked from the documentation I was reading, so I did as was written there. Makes no sense, give the variable is used only once